src/test/kotlin/org/rust/ide/hints/RustParameterInfoHandlerTest.kt

```diff
@@ -0,0 +1,175 @@
+        fn main() { foo(5, 32<caret>); }
+    """, "a1: u32, a2: u32", 1)
+
+    fun testFnArgTooManyArgs() = checkByText("""
+        fn foo(a1: u32, a2: u32) {}
+        fn main() { foo(0, 32,<caret>); }
+    """, "a1: u32, a2: u32", -1)
+
+    fun testFnClosure() = checkByText("""
+        fn foo(fun: Fn(u32) -> u32) {}
+        fn main() { foo(|x| x + <caret>); }
+    """, "fun: Fn(u32) -> u32", 0)
+
+    fun testFnNestedInner() = checkByText("""
+        fn add(v1: u32, v2: u32) -> u32 { v1 + v2 }
+        fn display(v: u32, format: &'static str) {}
+        fn main() { display(add(4, <caret>), "0.00"); }
+    """, "v1: u32, v2: u32", 1)
+
+    fun testFnNestedOuter() = checkByText("""
+        fn add(v1: u32, v2: u32) -> u32 { v1 + v2 }
+        fn display(v: u32, indent: bool, format: &'static str) {}
+        fn main() { display(add(4, 7), false, <caret>"); }
+    """, "v: u32, indent: bool, format: &'static str", 2)
+
+    fun testMultiline() = checkByText("""
+        fn sum(v1: u32, v2: u32, v3: u32) -> u32 { v1 + v2 + v3 }
+        fn main() {
+            sum(
+                10,
+                <caret>
+            );
+        }
+    """, "v1: u32, v2: u32, v3: u32", 1)
+
+    fun testAssocFn() = checkByText("""
+        struct Foo;
+        impl Foo { fn new(id: u32, val: f64) {} }
+        fn main() {
+            Foo::new(10, <caret>);
+        }
+    """, "id: u32, val: f64", 1)
+
+    fun testMethod() = checkByText("""
+        struct Foo;
+        impl Foo { fn bar(&self, id: u32, name: &'static name, year: u16) {} }
+        fn main() {
+            let foo = Foo{};
+            foo.bar(10, "Bo<caret>b", 1987);
+        }
+    """, "id: u32, name: &'static name, year: u16", 1)
+
+    fun testTraitMethod() = checkByText("""
+        trait Named {
+            fn greet(&self, text: &'static str, count: u16, l: f64);
+        }
+        struct Person;
+        impl Named for Person {
+            fn greet(&self, text: &'static str, count: u16, l: f64) {}
+        }
+        fn main() {
+            let p = Person {};
+            p.greet("Hello", 19, 10.21<caret>);
+        }
+    """, "text: &'static str, count: u16, l: f64", 2)
+
+    fun testNotArgs1() = checkByText("""
+        fn foo() {}
+        fn main() { fo<caret>o(); }
+    """, "", -1)
+
+    fun testNotArgs2() = checkByText("""
+        fn foo() {}
+        fn main() { foo()<caret>; }
+    """, "", -1)
+
+    fun testNotArgs3() = checkByText("""
+        fn foo(v<caret>: u32) {}
+    """, "", -1)
+
+    private fun checkByText(code: String, hint: String, index: Int) {
+        myFixture.configureByText("main.rs", code)
+        val handler = RustParameterInfoHandler()
+        val createContext = MockCreateParameterInfoContext(myFixture.editor, myFixture.file)
+
+        // Check hint
+        val elt = handler.findElementForParameterInfo(createContext)
+        if (hint.isNotEmpty()) {
+            elt ?: throw AssertionFailedError("Element not found")
```

Comment:
The idiom is to bind the result of `checkNotNull`:

```
val nonNullElt = checkNotNull(elt)
```

though, it'd be inconvenient here because of the shadowing :(